biz.aQute.bndlib/src/aQute/bnd/build/Run.java

```diff
@@ -1,18 +1,59 @@
 package aQute.bnd.build;
 
+import static java.lang.invoke.MethodType.methodType;
+
 import java.io.File;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.util.Map;
 
 import aQute.bnd.osgi.Processor;
+import aQute.lib.exceptions.Exceptions;
 
 public class Run extends Project {
 
+	/**
+	 * So far we have not included the resolver code inside bndlib. This means
+	 * that anybody inside bndlib cannot rely on the resolver code present.
+	 * However, we get more functions that take advantage of this tool. The
+	 * following code looks if the resolving counterpart of Run is present in
+	 * the classloader. Since this is a 1:1 replacement for this class, we try
+	 * to load the resolving Bndrun.
+	 */
+	private final static MethodHandle createBndrun;
+
+	static {
+		MethodHandles.Lookup lookup = MethodHandles.publicLookup();
+		MethodType type = methodType(Run.class, Workspace.class, File.class);
```

Comment:
Doesn't `Bndrun.createBndrun` return a `Bndrun` object? Does this find the method you are looking for since you are asking for a methodType returning `Run.class`? Perhaps you need to create the MethodType after you are able to load Bndrun.class using the class as the first argument instead of `Run.class`?

Reply:
Wow, I thought method types were intended to resolve compatible types? I added a test and you were right, it missed it.